src/components/Permissions/DataList.spec.jsx

```diff
@@ -17,13 +37,152 @@ jest.mock('components/PageTitle', () => {
+      data-secondary={secondary}
+    ></div>
+  )
+})
+
 describe('DataList', () => {
+  beforeEach(() => {
+    const queryResultApps = {
+      fetchStatus: 'loaded',
+      data: [
+        {
+          permissions: {
+            files: {
+              type: 'io.cozy.files',
+              description: 'Required to access the files'
+            },
+            allFiles: {
+              type: 'io.cozy.files.*',
+              description: 'Required to access the files'
+            },
+            apps: {
+              type: 'io.cozy.apps',
+              description:
+                'Required by the cozy-bar to display the icons of the apps',
+              verbs: ['GET']
+            }
+          },
+          slug: 'contacts',
+          name: 'Contacts'
+        },
+        {
+          permissions: {
+            files: {
+              type: 'io.cozy.files',
+              description: 'Required to access the files'
+            },
+            apps: {
+              type: 'io.cozy.apps',
+              description:
+                'Required by the cozy-bar to display the icons of the apps',
+              verbs: ['GET']
+            }
+          },
+          slug: 'home',
+          name: 'Home'
+        },
+        {
+          permissions: {
+            files: {
+              type: 'io.cozy.files',
+              description: 'Required to access the files'
+            }
+          },
+          slug: 'settings',
+          name: 'Settings'
+        },
+        {
+          permissions: {},
+          slug: 'store',
+          name: 'Store'
+        }
+      ]
+    }
+    const queryResultKonnectors = {
+      fetchStatus: 'loaded',
+      data: [
+        {
+          permissions: {
+            apps: {
+              type: 'io.cozy.apps',
+              description:
+                'Required by the cozy-bar to display the icons of the apps',
+              verbs: ['GET']
+            }
+          },
+          slug: 'alan',
+          name: 'Alan'
+        }
+      ]
+    }
+    isQueryLoading.mockReturnValue(true)
+    hasQueryBeenLoaded.mockReturnValue(true)
+    Q.mockReturnValue({ getById: () => 'kfrf' })
+    // useQuery.mockReturnValue(permissionsToDisplay)
+    useQuery.mockReturnValueOnce(queryResultApps)
+    useQuery.mockReturnValueOnce(queryResultKonnectors)
+    useQuery.mockReturnValueOnce(queryResultApps)
+    useQuery.mockReturnValueOnce(queryResultKonnectors)
+  })
+
+  it('should render a spinner when query is loading and has not been loaded', () => {
+    isQueryLoading.mockReturnValue(true)
+    hasQueryBeenLoaded.mockReturnValue(false)
+    const { queryByTestId } = render(<DataList />)
+    expect(queryByTestId('Spinner')).toBeTruthy()
+  })
+
   it('should display DataList title', () => {
-    const { getByText } = render(
-      <AppLike>
-        <DataList />
-      </AppLike>
+    const { queryByText } = render(<DataList />)
+    expect(queryByText('Permissions.data')).toBeTruthy
```

Comment:
did you check this test fail when Permissions data is not present?

I don't think so

Reply:
Does the stack accept to install apps or konnectors without permission? If yes, maybe the test can be useful, if not, not sure.